wagtail/contrib/table_block/tests.py

```diff
@@ -193,3 +200,163 @@ def test_render_with_extra_context(self):
+        translation.activate('it')
+        # Italian
+        block_it = TableBlock()
+        self.assertEqual('it', block_it.table_options['language'])
+        # table_options with language provided, different to envrionment
+        block_with_lang = TableBlock(table_options={'language': 'ja'})
+        self.assertNotEqual('it', block_with_lang.table_options['language'])
+        self.assertEqual('ja', block_with_lang.table_options['language'])
+        translation.activate('en')
+
+    def test_table_options_context_menu(self):
+        """
+        Test how contextMenu is set to default.
+        """
+        default_context_menu = list(DEFAULT_TABLE_OPTIONS['contextMenu'])  # create copy
+        # confirm the default is correct
+        table_options = TableBlock().table_options
+        self.assertEqual(table_options['contextMenu'], default_context_menu)
+        # confirm that when custom option is True, default is still used
+        table_options_menu_true = TableBlock(table_options={'contextMenu': True}).table_options
+        self.assertEqual(table_options_menu_true['contextMenu'], default_context_menu)
+        # confirm menu is removed if False is passed in
+        table_options_menu_false = TableBlock(table_options={'contextMenu': False}).table_options
+        self.assertEqual(table_options_menu_false['contextMenu'], False)
+        # confirm if list passed in, it is used
+        table_options_menu_list = TableBlock(table_options={'contextMenu': ['undo', 'redo']}).table_options
+        self.assertEqual(table_options_menu_list['contextMenu'], ['undo', 'redo'])
+        # test if empty array passed in
+        table_options_menu_list = TableBlock(table_options={'contextMenu': []}).table_options
+        self.assertEqual(table_options_menu_list['contextMenu'], [])
+
+    def test_table_options_others(self):
+        """
+        Test simple options overrides get passed correctly.
+        """
+        block_1_opts = TableBlock(table_options={'startRows': 5, 'startCols': 2}).table_options
+        self.assertEqual(block_1_opts['startRows'], 5)
+        self.assertEqual(block_1_opts['startCols'], 2)
+
+        block_2_opts = TableBlock(table_options={'stretchH': 'none'}).table_options
+        self.assertEqual(block_2_opts['stretchH'], 'none')
+
+        # check value that is not part of the defaults
+        block_3_opts = TableBlock(table_options={'allowEmpty': False}).table_options
+        self.assertEqual(block_3_opts['allowEmpty'], False)
+
+
+    def test_tableblock_render_form(self):
+        """
+        Test the rendered form field generated by TableBlock.
+        """
+        block = TableBlock()
+        html = block.render_form(value=self.value)
+        self.assertIn('<script>initTable', html)
+        self.assertIn('<div class="field char_field widget-table_input">', html)
+        # check that options render in the init function
+        self.assertIn(json.dumps(block.table_options), html)
```

Comment:
Will the keys be ordered consistently in the dumps? I'm worried the test might fail randomly (but if we're sure it doesn't, that's all fine).